synchronizer/models.py

```diff
@@ -498,6 +499,26 @@ def delete(cls, sync_id):
             db.session.commit()
         return True
 
+    def validate_worklogs(self):
+        worklogs = self.worklogs
+        target_name = self.target.connector_type.name
+
+        target_connector = ConnectorManager.create_connector(
+            target_name,
+            server=self.target.server,
+            api_token=self.target.api_token,
+            login=self.target.login,
+            password=self.target.password
+        )
+        
+        for worklog in worklogs:
+            if not target_connector.validate_issue(worklog.issue_id):
+                worklog.is_valid = False
+                db.session.add(worklog)

```

Comment:
I maybe wrong but I think that `session.add()` is only required for new records. Modified records can be committed without adding to session.